homeassistant/components/camera/dlink.py

```diff
@@ -0,0 +1,341 @@
+
+        upload_settings = BeautifulSoup(res.content)
+
+        u_settings = self.extract_form_fields(upload_settings)
+        self._is_ftp_upload_enabled = (
+            True if
+            u_settings.get('FTPScheduleEnable') == '1' else
+            False)
+
+        self._web_ui_form_data['upload'] = u_settings
+
+        self._ftp_host = u_settings.get('FTPHostAddress')
+        self._ftp_port = u_settings.get('FTPPortNumber')
+        self._ftp_username = u_settings.get('FTPUserName')
+        self._ftp_password = u_settings.get('FTPPassword')
+        self._ftp_relative_path = u_settings.get('FTPDirectoryPath')
+
+        ftp_comp = get_component('ftp')
+        if ftp_comp is not None and ftp_comp.FTP_SERVER is not None:
+            self._ftp_path = os.path.join(
+                ftp_comp.FTP_SERVER.ftp_root_path,
+                self._ftp_relative_path)
+
+        self._is_ftp_configured = self.check_ftp_settings()
+        self._lastconfig_update_from_device = datetime.datetime.now()
+        self.send_motion_state()
+        self.check_required_configurators()
+
+    def enable_motion_detection(self):
+        """ Enable the motion detection settings for the camera. """
+        # pylint: disable=missing-super-argument
+        can_enable = super().enable_motion_detection()
+        if can_enable is False:
+            return can_enable
+
+        self._web_ui_form_data['motion']['MotionDetectionEnable'] = 1
+        self._web_ui_form_data['motion']['MotionDetectionScheduleMode'] = 0
+        # The camera won't detect any motion if there are no blocks set
+        # so we default them to all set if none are selected
+        if (self._web_ui_form_data['motion']['MotionDetectionBlockSet'] ==
+                '0000000000000000000000000'):
+            self._web_ui_form_data['motion']['MotionDetectionBlockSet'] = (
+                '1111111111111111111111111')
+        requests.post(
+            self.base_url + 'setSystemMotion',
+            data=self._web_ui_form_data['motion'],
+            auth=(self.username, self.password))
+
+        # self._is_motion_detection_enabled = False
+        self.refesh_all_settings_from_device()
+        self.update_ha_state(True)
+
+        # self.send_motion_state()
+
+        return True
+
+    def disable_motion_detection(self):
+        """ Disable the motion detection settings for the camera. """
+        # pylint: disable=missing-super-argument
+        can_enable = super().disable_motion_detection()
+        if can_enable is False:
+            return can_enable
+
+        self._web_ui_form_data['motion']['MotionDetectionEnable'] = 0
+        requests.post(
+            self.base_url + 'setSystemMotion',
+            data=self._web_ui_form_data['motion'],
+            auth=(self.username, self.password))
+
+        self.refesh_all_settings_from_device()
+        self.update_ha_state(True)
+
+        # self.send_motion_state()
+
+        return True
+
+    def set_ftp_details(self):
```

Comment:
I think that this method should take in the parameters `server_ip`, `server_port`, `username`, `password`, `path`. That way it is not tightly coupled with our implementation of the ftp server.